pkg/cluster/clustermsi.go

```diff
@@ -137,3 +154,65 @@ func (m *manager) clusterMsiSecretName() (string, error) {
 
 	return fmt.Sprintf("%s-%s", m.doc.ID, clusterMsi.Name), nil
 }
+
+func (m *manager) clusterIdentityIDs(ctx context.Context) error {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return fmt.Errorf("clusterIdentityIDs called for CSP cluster")
+	}
+
+	clusterMsiResourceId, err := m.doc.OpenShiftCluster.ClusterMsiResourceId()
+	if err != nil {
+		return err
+	}
+
+	uaMsiRequest := dataplane.UserAssignedMSIRequest{
+		IdentityURL: m.doc.OpenShiftCluster.Identity.IdentityURL,
+		ResourceIDs: []string{clusterMsiResourceId.String()},
+		TenantID:    m.doc.OpenShiftCluster.Identity.TenantID,
+	}
+
+	msiCredObj, err := m.msiDataplane.GetUserAssignedIdentities(ctx, uaMsiRequest)
+	if err != nil {
+		return err
+	}
+
+	identity, err := getSingleExplicitIdentity(msiCredObj)
+	if err != nil {
+		return err
+	}
+	if identity.ClientID == nil || identity.ObjectID == nil {
+		return fmt.Errorf("unable to pull clientID and objectID from the MSI CredentialsObject")
+	}
+
+	clientId := *msiCredObj.CredentialsObject.ExplicitIdentities[0].ClientID
+	principalId := *msiCredObj.CredentialsObject.ExplicitIdentities[0].ObjectID
```

Comment:
WHy are we not using `identity.ClientId` here.

Reply:
Miss on my part, I'll fix this.

Reply:
Fixed